### ISSUE TITLE
cmd/compile: wrong evaluation order for map assignment [Go1.9]

### ISSUE THREAD

**Issue by randall77** [golang/go#23005]
Go 1.9.3 issue for #22881.

**Comment by JicLotus** [comment]
Can I take it?

**Comment by odeke-em** [comment]
Thank you @JicLotus for the offer, but this issue was already fixed by CL https://go-review.googlesource.com/81817 aka commit https://github.com/golang/go/commit/dd7cbf3a846c2cb125ac65173abaf6a8b9f903ff. @randall77 opened this issue as a placeholder for a backport to Go1.9, when Go1.9.3 is being released.

If you'd like to work on some compiler issues there are very many interesting ones https://github.com/golang/go/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+cmd%2Fcompile+in%3Atitle
or even generally all other issues, you are welcome to check out any, and thank you.

**Comment by JicLotus** [comment]
Many Thanks @odeke-em

**Comment by ianlancetaylor** [comment]
My opinion is that we should not fix this issue for 1.9.3 (or 1.8.6).  It may in some rare cases change the behavior of existing programs (and if it does not change the behavior, it won't do anything).  The behavior is a correct fix for the language, but it seems to me unlikely to actually fix a program that is currently working.  I think it's better to reserve this kind of change for 1.10 rather than run the risk of breaking programs in a minor release.

To put it another way: chance of breaking a program with 1.9.3: low.  Chance of fixing a program with 1.9.3: lower.  So, don't backport.

**Comment by randall77** [comment]
There is the case of future programs being written under 1.9, which would benefit from this bug being fixed.

But with that caveat, I still agree with Ian, probably not worth backporting.
The workaround for this issue is easy.

**Comment by ianlancetaylor** [comment]
Thanks, closing.
